### DISCUSSION TITLE
Injecting custom CSS files in each WebView

### DISCUSSION THREAD

**Discussion by 838** [readium/kotlin-toolkit#260] [Q&A] [answered]
We have created a custom CSS file for the comics book. What would be a good approach of injecting that CSS file into the each WebView?

**Comment by mickael-menu** [comment, marked as answer]
We plan to work soonish on an Injectable API to inject custom JavaScript and CSS but there's nothing available yet.

However, you can implement a custom transformer for the HTML resources to use with the `TransformingFetcher` and inject the CSS yourself.

You can wrap the publication fetcher from the `Streamer.open(onCreatePublication:)` closure:

```kotlin
suspend fun injectCss(resource: Resource): Resource {
    ...
}

streamer.open(
    ...,
    onCreatePublication = {
        fetcher = TransformingFetcher(fetcher, injectCss)
    }
)
```

Take a look at the `HtmlInjector` for an example:

* https://github.com/readium/kotlin-toolkit/blob/540c80fc1c57f04d09db45764f103724177aedf9/readium/streamer/src/main/java/org/readium/r2/streamer/fetcher/HtmlInjector.kt#L27

**Reply by 838** [reply]
@mickael-menu Thanks for your help 🙏 🙂
